pslab/external/MLX90614.py

```diff
@@ -1,46 +1,40 @@
-from __future__ import print_function
+from pslab.bus import I2CSlave
 
-
-def connect(route, **args):
-    return MLX90614(route, **args)
-
-
-class MLX90614():
+class MLX90614(I2CSlave):
+    _ADDRESS = 0x5A
+    _OBJADDR = 0x07
+    _AMBADDR = 0x06
     NUMPLOTS = 1
     PLOTNAMES = ['Temp']
-    ADDRESS = 0x5A
     name = 'PIR temperature'
 
-    def __init__(self, I2C, **args):
-        self.I2C = I2C
-        self.ADDRESS = args.get('address', self.ADDRESS)
-        self.OBJADDR = 0x07
-        self.AMBADDR = 0x06
+    def __init__(self):
+        super().__init__(self._ADDRESS)
 
-        self.source = self.OBJADDR
+        self.source = self._OBJADDR
 
         self.name = 'Passive IR temperature sensor'
         self.params = {'readReg': {'dataType': 'integer', 'min': 0, 'max': 0x20, 'prefix': 'Addr: '},
                        'select_source': ['object temperature', 'ambient temperature']}
 
-        try:
```

Comment:
If the sensor works with the default baudrate (125 kHz), there is no need for this configuration code.

Even if the sensor requires the baudrate to not exceed some maximum value, merely instantiating the sensor should not automatically reconfigure the I2C bus, I think. The user might have multiple sensors connected to the I2C bus and have manually set the baudrate to a desired speed.

Instead we should check if the baudrate is low enough for this sensor, and raise a warning or even an exception if it is too high. However, there is currently no way to read back the configured I2C baudrate (#195).